lib/awful/screen.lua

```diff
@@ -505,6 +564,100 @@ function screen.object.get_selected_tag(s)
     return screen.object.get_selected_tags(s)[1]
 end
 
+local function normalize(ratios, size)
+    local sum = 0
+
+    for _, r in ipairs(ratios) do
+        sum = sum + r
+    end
+
+    -- Avoid to mutate the input.
+    local ret = {}
+    local sum2 = 0
+
+    for k, r in ipairs(ratios) do
+        ret[k] = (r*100)/sum
+        ret[k] = math.floor(size*ret[k]*0.01)
+        sum2 = sum2 + ret[k]
+    end
+
+    -- Fix the sub-pixel issue.
```

Comment:
No idea what "the sub-pixel issue" is, so this comment is (to me) useless. Please either clarify or just remove it.

Reply:
Screens geometry are integers, ratios are floats.